Upgrade a Chain tutorial (#670)
substrate-developer-hub/substrate-developer-hub.github.io: docs/tutorials/upgrade-a-chain/index.md

---
title: Introduction
---

One of the defining features of the Substrate blockchain development framework is its support for
forkless runtime upgrades. This means that Substrate-based networks can be enhanced in a way that is
supported and protected by the capabilities of the blockchain itself. A blockchain's
[runtime](../../knowledgebase/runtime) is the mechanism by which

> [state](../../knowledgebase/runtime/storage) changes are effected. Because the definition of the

Reviewer: Two gross sentences. "This ____ means...", "enhanced in a way that is", "changes are effected."

Author: I did my best to address this comment, but next time something a little more constructive/specific would be appreciated :pray:

Reviewer: It's better, I would still try to be more active. "A runtime defines the information that your blockchain can hold and the logic for effecting changes to that information. Substrate supports the capability to make breaking changes to your runtime without a hard fork." 

Then you say _how_ it supports that capability: "Because the definition of the runtime is itself an element in a Substrate chain's state..."

Author: "Active voice" is a writing concept that ISWTG I _cannot_ wrap my head around :sweat_smile:

mechanisms for forkless upgrades of [FRAME](../../knowledgebase/runtime/frame)-based runtimes.
First, the
[`sudo_unchecked_weight`](https://substrate.dev/rustdocs/v2.0.0-rc6/pallet_sudo/enum.Call.html#variant.sudo_unchecked_weight)
function from the [Sudo pallet](../../knowledgebase/runtime/frame#sudo) will be used to perform an
upgrade that adds the [Scheduler pallet](../../knowledgebase/runtime/frame#scheduler). Then, the
[`schedule`](https://substrate.dev/rustdocs/v2.0.0-rc6/pallet_scheduler/enum.Call.html#variant.schedule)
function from the Scheduler pallet will be used to perform an upgrade that increases the
[existential (minimum) balance](../../knowledgebase/getting-started/glossary#existential-deposit)
for network accounts.

If you have problems with this tutorial, the Substrate community is full of helpful resources. We
maintain an active
[#SubstrateTechnical chat room](https://app.element.io/#/room/!HzySYSaIhtyWrwiwEV:matrix.org) and
monitor the
[`substrate` tag on Stack Overflow](https://stackoverflow.com/questions/tagged/substrate). You can
also use the [`subport` GitHub repository](https://github.com/paritytech/subport/issues/new) to

> create an issue. We encourage community to be actively part of the future of Substrate. 

Reviewer: Active verbs when possible
```suggestion
create an issue. We encourage the community to actively participate in the future of Substrate. 
```
